src/test/scala/edu/ie3/simona/agent/grid/DBFSAlgorithmCenGridSpec.scala

```diff
@@ -130,48 +120,37 @@ class DBFSAlgorithmCenGridSpec
         )
 
       val key =
-        ScheduleLock.singleKey(TSpawner, scheduler.ref.toTyped, INIT_SIM_TICK)
-      scheduler.expectMsgType[ScheduleActivation] // lock activation scheduled
+        ScheduleLock.singleKey(TSpawner, scheduler.ref, INIT_SIM_TICK)
+      scheduler
+        .expectMessageType[ScheduleActivation] // lock activation scheduled
 
-      centerGridAgent ! GridAgent.Create(
+      centerGridAgent ! CreateGridAgent(
         gridAgentInitData,
         key,
       )
-      scheduler.expectMsg(
-        ScheduleActivation(centerGridAgent.toTyped, INIT_SIM_TICK, Some(key))
-      )
 
-      scheduler.send(centerGridAgent, Activation(INIT_SIM_TICK))
-      scheduler.expectMsg(
-        Completion(
-          centerGridAgent.toTyped,
-          Some(3600),
-        )
-      )
+      val msg = scheduler.expectMessageType[ScheduleActivation]
+      msg shouldBe ScheduleActivation(msg.actor, INIT_SIM_TICK, Some(key))
```

Comment:
You could also use `expectMessage` here as before?

Reply:
That is not possible, because the work around ``msg.actor`` is necessary.

Reply:
True, but you can save the Activation ActorRef in a `val` the first time and use it subsequently in each test case. So in for this example, this is how I would have solved it before:

```scala
      val msg = scheduler.expectMessageType[ScheduleActivation]
      msg.tick shouldBe INIT_SIM_TICK
      msg.unlockKey shouldBe Some(key)
      val gridAgentActivation = msg.actor

      centerGridAgent ! ActivationAdapter(Activation(INIT_SIM_TICK))
      scheduler.expectMessage(Completion(gridAgentActivation, Some(3600)))
```